tasks/abs.rb

```diff
@@ -75,8 +75,13 @@ def provision(platform, inventory_location, vars)
   data.each do |host|
     if platform_uses_ssh(host['type'])
       node = { 'uri' => host['hostname'],
-               'config' => { 'transport' => 'ssh', 'ssh' => { 'user' => ENV['ABS_USER'], 'password' => ENV['ABS_PASSWORD'], 'host-key-check' => false } },
+               'config' => { 'transport' => 'ssh', 'ssh' => { 'user' => ENV['ABS_USER'], 'host-key-check' => false } },
                'facts' => { 'provisioner' => 'abs', 'platform' => host['type'], 'job_id' => job_id } }
+      if !ENV['ABS_SSH_PRIVATE_KEY'].empty?
```

Comment:
@jpartlow this throws an error when the environment variable is unset.

Reply:
This happens when merging goes fast forward. We need to come with a proper fix in order to avoid blocking errors.

Reply:
Sorry about that; #185 is merged which should fix the problem I introduced, but I added #186 as a beginning for some spec coverage for the module tasks.